### PR TITLE
feat: add lint for conda forge specific stuff

### PR DESCRIPTION
**Pull request by nichmor**
Checklist
* [ ] Added a ``news`` entry
* [ ] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -432,14 +453,15 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
 
     # 4: Do not delete example recipe
     if is_staged_recipes and recipe_dir is not None:
+        recipe_name = "meta.yaml" if not is_rattler_build else "recipe.yaml"
```
Comment (wolfv) [inline conda_smithy/lint_recipe.py:456]: this should not depend on `is_rattler_build`.

Comment (nichmor) [reply conda_smithy/lint_recipe.py]: agree, done

File: news/linting.rst [hunk 2]
```diff
@@ -1,6 +1,5 @@
 **Added:**
-
-* Added more lints / hints for conda v2 recipes. (#2000, #2003, #2008, #2016)
+* Added more lints / hints for conda v2 recipes. (#2000, #2008, #2022)
```
Comment (beckermr) [inline news/linting.rst:2]:
```suggestion
* Added more lints / hints for conda v2 recipes. (#2000, #2003, #2008, #2016, #2022)
```

**Review by beckermr — CHANGES_REQUESTED** [review]
One nit.

**Review by beckermr — APPROVED** [review]

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -451,32 +474,39 @@ def run_conda_forge_specific(meta, recipe_dir, lints, hints):
...
-            if msg not in lints:
-                lints.append(msg)
+            if not os.path.exists(example_fname):
+                msg = (
+                    "Please do not delete the example recipe found in "
+                    f"`recipes/example/{recipe_name}`."
+                )
+
+                if msg not in lints:
+                    lints.append(msg)
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:489]: This is wrong. There's no `example/recipe.yaml` to start with.

**Pull request merged**